Based on https://github.com/ofiwg/libfabric/pull/7583 by mwheinz
Updated for release version 1.15.1 by Jasper Grimm
----------------------------------------------------------------
diff -Nru libfabric-1.15.1.orig/prov/opx/configure.m4 b/prov/opx/configure.m4
--- libfabric-1.15.1.orig/prov/opx/configure.m4	2022-05-16 13:14:58.000000000 +0100
+++ b/prov/opx/configure.m4	2022-05-16 13:25:28.000000000 +0100
@@ -44,7 +44,13 @@
 	opx_happy=0
 	opx_direct=0
 
-	AS_IF([test x"$enable_opx" != x"no"],[
+
+	dnl OPX hardware is not supported for MacOS or FreeBSD,
+	dnl and is not supported for non-x86 processors.
+	AS_IF([test "x$macos" = "x1"],[opx_happy=0],
+		[test "x$freebsd" = "x1"],[opx_happy=0],
+		[test x$host_cpu != xx86_64],[opx_happy=0],
+		[test x"$enable_opx" != x"no"],[
 
 		AC_MSG_CHECKING([for opx provider])
 
@@ -91,6 +97,7 @@
 		AC_SUBST(opx_reliability, [$OPX_RELIABILITY])
 		AC_DEFINE_UNQUOTED(OPX_RELIABILITY, [$OPX_RELIABILITY], [fabric direct reliability])
 
+		opx_happy=1
 		FI_CHECK_PACKAGE([opx_uuid],
 			[uuid/uuid.h],
 		   	[uuid],
@@ -98,14 +105,19 @@
 		   	[],
 		   	[],
 		   	[],
-		   	[opx_happy=1],
+		   	[],
+		   	[opx_happy=0])
+		FI_CHECK_PACKAGE([opx_numa],
+			[numa.h],
+		   	[numa],
+		   	[numa_node_of_cpu],
+		   	[],
+		   	[],
+		   	[],
+		   	[],
 		   	[opx_happy=0])
 
-		dnl OPX hardware is not available for MacOS or FreeBSD.
-		AS_IF([test "x$macos" = "x1"],[opx_happy=0],[])
-		AS_IF([test "x$freebsd" = "x1"],[opx_happy=0],[])
-
-		AC_CHECK_DECL([HAVE_ATOMICS],
+               AC_CHECK_DECL([HAVE_ATOMICS],
                              [],
                              [cc_version=`$CC --version | head -n1`
                               AC_MSG_WARN(["$cc_version" doesn't support native atomics.  Disabling OPX provider.])
